src/cuga/backend/server/manage_routes.py

```
@@ -1427,115 +1822,93 @@ async def patch_draft_knowledge(request: Request, agent_id: Optional[str] = None
+        # ``logger.error(f"...: {e}")`` swallowed everything when
+        # ``str(e)`` was empty (some libs raise bare Exception() with
+        # no args) and left us no breadcrumb to diagnose. Include
+        # repr(e) so we at least see the class name when str is empty.
+        # Full detail (incl. embedding-API errors, paths, partial key material)
+        # goes to the LOG only; the client gets a generic message (Sami review /
+        # CodeQL — don't leak internals in the HTTP body).
+        logger.exception(f"Failed to patch draft knowledge: {e!r}")
+        raise HTTPException(
+            status_code=500,
+            detail="Failed to save knowledge settings. Check the server logs for details.",
+        )
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)